rust/xaynet-analytics/src/controller.rs

```diff
@@ -121,17 +146,20 @@ impl AnalyticsController {
         )
     }
 
-    // TODO: review and debug this method during https://xainag.atlassian.net/browse/XN-1560
```

Comment:
After having talked with @acrrd , i will add a comment in the code to explain why this is implemented as it is and not just checking whether `last_time_data_sent > Utc::now() - self.send_data_frequency`.

The reason is that i wanted to have sort of "bins" so that the data can be sent only once per "bin", and not more dynamically as the simpler alternative above. 

So that if for example `self.send_data_frequency = Duration::hours(6)`, and the last time we sent the data was at 5AM, we would be able to send again at 7AM, while with the simpler solution we wouldn't be able to send again until 11AM. 

The reason i went for the current solution is that it might be easier on the coordinator side (and then in the UI) to aggregate data for fixed bins, instead of more dynamically "each `self.send_data_frequency` hours". 

But the mental model I had was about sending each 24h, when in that case it does make sense to only send once between midnight and midnight UTC.

But honestly I'm not sure what's the "correct" approach, it highly highly depends on the volume of data so that each aggregation has enough data in it for the math to work out. Maybe a "sufficient volume", however that is calculated could be achieved only if sending once a week (in that case the hardcoded `MAX_SEND_DATA_FREQUENCY_HOURS` of 24 hours should be revisited, but it shouldn't be too difficult). 

I guess there will just need to be some experiments to see what works or whatnot, or some very advanced features that set this threshold dynamically based on data volume.

As said i'll add this comment in the code when addressing the other review comments.

Reply:
ok added